controllers/sessionController.js

```diff
@@ -0,0 +1,49 @@
+const User = require("../models/User");
+const parseVErr = require("../utils/parseValidationErrs");
+
+const registerShow = (req, res) => {
+  res.render("register");
+};
+
+const registerDo = async (req, res, next) => {
+  if (req.body.password != req.body.password1) {
+    req.flash("error", "The passwords entered do not match.");
+    res.render("register");
```

Comment:
You need to pass the error as a parameter here in order for the errors to be displayed in the front end.
Do this:
```
res.render("register",{errors: req.flash("error")});
```